storage/test/transfer.test.js

```diff
@@ -0,0 +1,639 @@
+      patch: sinon.stub().callsArgWith(1, null, transferJobMock),
+      list: sinon.stub().callsArgWith(1, null, { transferJobs: [transferJobMock] })
+    },
+    transferOperations: {
+      get: sinon.stub().callsArgWith(1, null, transferOperationMock),
+      pause: sinon.stub().callsArgWith(1, null, transferOperationMock),
+      resume: sinon.stub().callsArgWith(1, null, transferOperationMock),
+      list: sinon.stub().callsArgWith(1, null, { operations: [transferOperationMock] })
+    }
+  };
+  var googleapisMock = {
+    storagetransfer: sinon.stub().returns(storagetransferMock),
+    auth: {
+      getApplicationDefault: sinon.stub().callsArgWith(0, null, {})
+    }
+  };
+  return {
+    program: proxyquire('../transfer', {
+      googleapis: googleapisMock
+    }),
+    mocks: {
+      googleapis: googleapisMock,
+      storagetransfer: storagetransferMock,
+      transferJob: transferJobMock,
+      transferOperation: transferOperationMock
+    }
+  };
+}
+
+describe('storage:transfer', function () {
+  describe('jobs', function () {
+    describe('create', function () {
+      it('should create a transfer job', function () {
+        var description = 'description';
+        var sample = getSample();
+        var callback = sinon.stub();
+
+        sample.program.createTransferJob(bucketName, bucketName, '2016/08/11', '15:30', null, callback);
+
+        assert(callback.calledOnce, 'callback called once');
+        assert.equal(callback.firstCall.args.length, 2, 'callback received 2 arguments');
+        assert.ifError(callback.firstCall.args[0], 'callback did not receive error');
+        assert.strictEqual(callback.firstCall.args[1], sample.mocks.transferJob, 'callback received transfer job');
+
+        sample.program.createTransferJob(bucketName, bucketName, '2016/08/11', '15:30', description, callback);
+
+        assert(callback.calledTwice, 'callback called twice');
+        assert.equal(callback.secondCall.args.length, 2, 'callback received 2 arguments');
+        assert.ifError(callback.secondCall.args[0], 'callback did not receive error');
+        assert.strictEqual(callback.secondCall.args[1], sample.mocks.transferJob, 'callback received transfer job');
+        assert.equal(sample.mocks.storagetransfer.transferJobs.create.secondCall.args[0].resource.description, description, 'description was set');
+      });
+
+      it('should require "srcBucketName"', function () {
+        var sample = getSample();
+        var callback = sinon.stub();
+
+        sample.program.createTransferJob(null, null, null, null, null, callback);
+
+        assert(callback.calledOnce, 'callback called once');
+        assert.equal(callback.firstCall.args.length, 1, 'callback received 1 argument');
+        assert(callback.firstCall.args[0], 'callback received error');
+        assert.equal(callback.firstCall.args[0].message, '"srcBucketName" is required!', 'error has correct message');
```

Comment:
Because JavaScript, e.g.

```js
var a = new Error('foo');
var b = new Error('foo');
a == b; // false
a.message == b.message; // true
```
Can't assert non-strict equality on instances of `Error`